src/agent/Core/Controller/CheckoutSession.cpp

```diff
@@ -178,36 +179,110 @@ Controller::maybeSend100Continue(Client *client, Request *req) {
 	}
 }
 
+void
+Controller::onTimeout(EV_P_ ev_timer *io, int flag) {
+	Request *req = static_cast<Request *>(io->data);
+	Client *client = static_cast<Client *>(req->client);
+	Controller *that = static_cast<Controller *>(Controller::getServerFromClient(client));
+    ev_io_stop(that->getLoop(), &req->connectedWatcher);
+	SystemException e("Waiting on socket connect timed out", ETIMEDOUT);
```

Comment:
You're creating an exception here so this function should have a TRACE_POINT() for better backtraces.